llama_index/embeddings/huggingface_utils.py

```diff
@@ -72,3 +74,26 @@ def format_text(
     # NOTE: strip() enables backdoor for defeating instruction prepend by
     # passing empty string
     return f"{instruction} {text}".strip()
+
+
+def get_pooling_mode(model_name: Optional[str]) -> str:
+    pooling_config_url = (
+        f"https://huggingface.co/{model_name}/raw/main/1_Pooling/config.json"
+    )
+
+    try:
+        response = requests.get(pooling_config_url)
+        config_data = response.json()
+
+        cls_token = config_data.get("pooling_mode_cls_token", False)
+        mean_tokens = config_data.get("pooling_mode_mean_tokens", False)
```

Comment:
so every model card will have those 2 flag with the same name always?

Reply:
pooling_mode_cls_token, pooling_mode_mean_tokens are two different names right.

Reply:
okay. Yeah, the names will be consistent for every model card.

Reply:
I mean does the model card for every model from huggingface always use `pooling_mode_cls_token` for cls pooling  and `pooling_mode_mean_tokens ` for mean pooling.
is it possible some models using different naming?

Reply:
Nope. Every model uses the same naming convention.

Reply:
ic good to know!